local/core/getDevices.go

```diff
@@ -41,13 +82,18 @@ func newDiscoveryHandler(
 }
 
 type discoveryHandler struct {
-	deviceCfg deviceConfiguration
-	handler   DeviceHandler
+	deviceCfg             deviceConfiguration
+	handler               DeviceHandlerV2
+	newDeviceWasSetCalled sync.Map
```

Comment:
I'm not sure what does the name mean in this context. Can you please explain?

Reply:
Fixed by rename property to `filterDiscoveredDevices`. It serves to call handler just once per device(deviceID). Otherwise it is not effective because it gets same data from the device multiple times.